libvips/foreign/webpsave.c

```diff
@@ -655,12 +662,15 @@ vips_foreign_save_webp_build( VipsObject *object )
 		return( -1 );
 	}
 
-	page_height = vips_image_get_page_height( webp->image );
-
 	/* RGB(A) frame as a contiguous buffer.
 	 */
-	webp->frame_bytes = g_malloc( (size_t) webp->image->Bands *
+	webp->frame_bytes = g_try_malloc( (size_t) webp->image->Bands *
 		webp->image->Xsize * page_height );
+	if( webp->frame_bytes == NULL ) {
+		vips_error( "webpsave", _( "out of memory" ) );
```

Comment:
If this is some sort of log, I'd add the amount of memory being requested to add context, something like (pseudocode): `vips_error( "webpsave", _( "can't allocate (webp->image->Bands * webp->image->Xsize * page_height) of memory" ) );`

Reply:
Sounds good 👍
I added this with 989b74c:
```
$ vips black out.webp 10000000 10000000                                
webpsave: failed to allocate 300000000000000 bytes
```